logbook-jaxws/src/test/java/org/zalando/logbook/jaxws/LoggingHandlerTest.java

```diff
@@ -0,0 +1,166 @@
+		verify(context).get(BindingProvider.ENDPOINT_ADDRESS_PROPERTY);
+		verify(context, times(3)).get(SOAPMessage.CHARACTER_SET_ENCODING);
+		verify(context).get(SOAPMessageContext.QUERY_STRING);
+		verify(context, times(9)).getMessage();
+	}
+
+	@Test
+	@Order(2)
+	void hanlder_with_soap_11_response() throws IOException, SOAPException {
+
+		// -- mocks
+		when(writer.isActive()).thenReturn(true);
+		when(context.get(MessageContext.MESSAGE_OUTBOUND_PROPERTY)).thenReturn(false);
+		lenient().when(context.getMessage()).thenReturn(SOAPMessageFactory.createSoap1_1());
+		lenient().when(context.get(MessageContext.HTTP_RESPONSE_CODE)).thenReturn(200);
+
+		// -- underTest
+		boolean result = underTest.handleMessage(context);
+
+		// -- assertions
+		assertTrue(result);
+		verify(context).get(MessageContext.MESSAGE_OUTBOUND_PROPERTY);
+		verify(context, times(9)).getMessage();
+		verify(context, Mockito.times(2)).get(MessageContext.HTTP_RESPONSE_CODE);
+	}
+
+	@Test
+	void handler_with_other_methods() {
+
+		// -- underTest
+		boolean fault = underTest.handleFault(context);
+		Set<QName> headers = underTest.getHeaders();
+		underTest.close(context);
+		
+		//-- assertions
+		assertFalse(fault);
+		assertEquals(headers, null);
+	}
+
+	@Test
+	void hanlder_with_soap_request_throw_exception() throws IOException {
+
+		// -- mocks
+		Logbook mockLogbook = mock(Logbook.class);
+		when(context.get(MessageContext.MESSAGE_OUTBOUND_PROPERTY)).thenReturn(true);
+		when(mockLogbook.process(any())).thenThrow(new IOException("ERROR"));
+
+		// -- underTest
+		LoggingHandler underTest = new LoggingHandler(mockLogbook);
+		RuntimeException exception = assertThrows(RuntimeException.class, () -> {
+			underTest.handleMessage(context);
+		});
+
+		// -- assertions
+		verify(context).get(MessageContext.MESSAGE_OUTBOUND_PROPERTY);
+		verify(mockLogbook).process(any());
+		assertTrue(exception.getCause().getClass() == IOException.class);
+	}
+
+	@Test
+	void hanlder_with_soap_response_throw_exception() throws NoSuchFieldException, SecurityException, Exception {
+		
+		// -- mocks
+		ResponseProcessingStage responseProcessingMock = mock(ResponseProcessingStage.class);
+		@SuppressWarnings("unchecked")
+		ThreadLocal<ResponseProcessingStage> stateMock = mock(ThreadLocal.class);
+		setFinalStatic(LoggingHandler.class.getDeclaredField("STAGE_CONTEXT"), stateMock);
+		when(context.get(MessageContext.MESSAGE_OUTBOUND_PROPERTY)).thenReturn(false);
+		when(stateMock.get()).thenReturn(responseProcessingMock);
+		when(responseProcessingMock.process(Mockito.any())).thenThrow(new IOException("ERROR"));
+
+		
+		// -- underTest
+		LoggingHandler underTest = new LoggingHandler(null);
+		RuntimeException exception = assertThrows(RuntimeException.class, () -> {
+			underTest.handleMessage(context);
+		});
+
+		// -- assertions
+		verify(context).get(MessageContext.MESSAGE_OUTBOUND_PROPERTY);
+		verify(stateMock).get();
+		assertTrue(exception.getCause().getClass() == IOException.class);
```

Comment:
If you want to use AssertJ fluent api:
```suggestion
		assertThat(xception.getCause()).isInstanceOf(IOException.class);
```